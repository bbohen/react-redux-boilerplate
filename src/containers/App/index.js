import styled from 'styled-components';

const App = styled.div`
  display: flex;
  flex-direction: column;
  font-size: 1em;
  background-color: ${props => props.theme.colors.background}
  min-height: 100vh;
`;

export default App;
