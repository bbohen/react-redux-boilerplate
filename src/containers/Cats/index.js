import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { connect } from 'react-redux';

import { Card, Heading } from 'components';
import Wrapper from './Wrapper';
import { load as loadCats } from '../../redux/modules/cats';

export class UnconnectedCats extends Component {
  static propTypes = {
    catsAreLoaded: PropTypes.bool.isRequired,
    loadCats: PropTypes.func.isRequired,
    cats: PropTypes.arrayOf(PropTypes.shape),
  }

  static defaultProps = {
    cats: [],
  }

  componentWillMount() {
    const { catsAreLoaded } = this.props;

    if (!catsAreLoaded) {
      this.props.loadCats();
    }
  }

  render() {
    const { cats } = this.props;

    return (
      <Wrapper>
        {cats.length ? cats.map(({ data }, index) => (
          <Card
            index={index}
            key={data.id}
            imageUrl={data.thumbnail}
          />
        )) :
        <Heading>Loading you some cats</Heading>}
      </Wrapper>
    );
  }
}

export function mapStateToProps(state) {
  return {
    cats: state.cats.list,
    catsAreLoaded: state.cats.isLoaded,
  };
}

export default connect(mapStateToProps, { loadCats })(UnconnectedCats);
