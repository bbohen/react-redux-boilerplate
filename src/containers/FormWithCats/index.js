import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';
import { Field, reduxForm } from 'redux-form';

import { FormField } from 'components';
import { clearRandomCat, giveRandomCat } from '../../redux/modules/cats';
import validation from './validation';
import ActionText from './ActionText';
import CatFormHeading from './Heading';
import CatFormButton from './Button';
import Form from './Form';
import Wrapper from './Wrapper';

const UnconnectedFormWithCats = (props) => {
  const { email, number, required } = validation;
  let result;

  if (props.randomCat) {
    result = (
      <div>
        <CatFormHeading>
          Here is your cat.
          <ActionText onClick={props.clearRandomCat}>
            Try again?
          </ActionText>
        </CatFormHeading>
        <img
          alt=""
          src={props.randomCat}
        />
      </div>);
  }

  return (
    <Wrapper>
      {!result &&
      <Form onSubmit={props.handleSubmit && props.handleSubmit(props.giveRandomCat)}>
        <CatFormHeading>Form.</CatFormHeading>
        <Field
          label="Name:"
          name="name"
          component={FormField}
          type="text"
          validate={required}
        />
        <Field
          label="Email:"
          name="email"
          component={FormField}
          type="text"
          validate={[email, required]}
        />
        <Field
          label="Number:"
          name="number"
          component={FormField}
          type="text"
          validate={[number, required]}
        />
        <CatFormButton type="submit">
            Submit the form.
        </CatFormButton>
      </Form> }
      {result}
    </Wrapper>
  );
};

UnconnectedFormWithCats.propTypes = {
  clearRandomCat: PropTypes.func.isRequired,
  giveRandomCat: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  randomCat: PropTypes.string,
};

UnconnectedFormWithCats.defaultProps = {
  randomCat: '',
};

export default connect(
  ({ cats = {} }) => ({
    randomCat: cats.randomCat,
  }),
  {
    clearRandomCat,
    giveRandomCat,
  },
)(reduxForm({
  form: 'cats',
})(UnconnectedFormWithCats));
