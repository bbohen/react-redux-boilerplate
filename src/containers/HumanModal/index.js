import PropTypes from 'prop-types';
import React from 'react';
import styled from 'styled-components';

import EnvelopeIcon from 'react-icons/lib/fa/envelope';
import BirthdayIcon from 'react-icons/lib/fa/birthday-cake';
import PhoneIcon from 'react-icons/lib/fa/phone';

import { Heading, Modal } from 'components';
import Details from './Details';
import Header from './Header';
import Icon from './Icon';
import Image from './Image';

const ModalHeaderHeading = styled(Heading)`
  margin-bottom: .5em;
`;

const HumanModal = ({ age, bio, birthday, imageUrl, email, name, phone, onClick, surname }) =>
  (<Modal onClick={onClick}>
    <Header>
      <Image
        alt=""
        src={imageUrl}
      />
      <Details>
        <ModalHeaderHeading>{name}, {surname}</ModalHeaderHeading>
        <div>
          <Icon>
            <EnvelopeIcon />
          </Icon>
          {email}
        </div>
        <div>
          <Icon>
            <BirthdayIcon />
          </Icon>
          {birthday.mdy} ({age} years old)
        </div>
        <div>
          <Icon>
            <PhoneIcon />
          </Icon>
          {phone}
        </div>
      </Details>
    </Header>
    <div>{bio}</div>
  </Modal>);

HumanModal.propTypes = {
  onClick: PropTypes.func.isRequired,
  age: PropTypes.number,
  bio: PropTypes.string,
  birthday: PropTypes.shape({
    mdy: PropTypes.string,
  }),
  email: PropTypes.string,
  imageUrl: PropTypes.string,
  name: PropTypes.string,
  phone: PropTypes.string,
  surname: PropTypes.string,
};

HumanModal.defaultProps = {
  age: 0,
  bio: '',
  birthday: {
    mdy: 'XX/XX/XXXX',
  },
  email: '',
  imageUrl: '',
  name: '',
  phone: '',
  surname: '',
};
export default HumanModal;
