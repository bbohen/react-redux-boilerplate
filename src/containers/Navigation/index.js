import React, { PropTypes } from 'react';

import { Heading } from 'components';
import { SECONDARY_COLOR } from 'theme';

import Wrapper from './Wrapper';
import Content from './Content';
import GitHubLink from './GitHubLink';
import List from './List';
import Item from './Item';
import LogoLink from './LogoLink';
import NavLink from './NavLink';

const activeStyle = {
  textShadow: `-1px -1px 3px ${SECONDARY_COLOR}`,
};

const Navigation = () =>
  (<Wrapper>
    <Content>
      <Heading>
        <LogoLink
          activeStyle={activeStyle}
          isActive={({ pathname }) => pathname === '/'}
          to="/"
        >
          React Redux Boilerplate
        </LogoLink>
      </Heading>
      <List>
        <Item>
          <NavLink
            activeStyle={activeStyle}
            isActive={({ pathname }) => pathname === '/cats'}
            to="/cats"
          >
            Cats
          </NavLink>
        </Item>
        <Item>
          <NavLink
            activeStyle={activeStyle}
            isActive={({ pathname }) => pathname === '/humans'}
            to="/humans"
          >
            Humans
          </NavLink>
        </Item>
        <Item>
          <NavLink
            activeStyle={activeStyle}
            isActive={({ pathname }) => pathname === '/form'}
            to="/form"
          >
            Form
          </NavLink>
        </Item>
      </List>
      <GitHubLink />
    </Content>
  </Wrapper>);

const contextTypes = {
  router: PropTypes.object,
};

Navigation.contextTypes = contextTypes;

export default Navigation;
