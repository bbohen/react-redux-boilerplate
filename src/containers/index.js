export App from './App';
export Cats from './Cats';
export ErrorBoundary from './ErrorBoundary';
export FormWithCats from './FormWithCats';
export Home from './Home';
export MainContent from './MainContent';
export Navigation from './Navigation';
