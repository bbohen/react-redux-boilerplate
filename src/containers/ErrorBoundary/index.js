import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { App, MainContent } from 'containers';
import { Heading } from 'components';

const ErrorHeading = Heading.extend`
  margin-bottom: 40px;
`;

export default class ErrorBoundary extends Component {
  static propTypes = {
    children: PropTypes.element.isRequired,
  }

  constructor(props) {
    super(props);
    this.state = {
      error: null,
      info: null,
      hasError: false,
    };
  }

  componentDidCatch(error, info) {
    this.setState({ hasError: true, error, info });
  }

  render() {
    if (this.state.hasError) {
      return (
        <App>
          <MainContent>
            <div>
              <ErrorHeading>
              Whoops!
              </ErrorHeading>
              <div>{this.state.error.name}</div>
              <div>{this.state.error.message}</div>
              <div>{this.state.error.type}</div>
              <div>{this.state.error.arguments}</div>
            </div>
          </MainContent>
        </App>
      );
    }
    return this.props.children;
  }
}
