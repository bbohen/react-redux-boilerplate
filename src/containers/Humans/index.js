import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { connect } from 'react-redux';

import { Card, Heading } from 'components';
import Wrapper from './Wrapper';
import HumanModal from '../HumanModal';
import { load as loadHumans, select as selectHuman } from '../../redux/modules/humans';
import { load as loadLoremIpsum } from '../../redux/modules/loremIpsum';

export class UnconnectedHumans extends Component {
  static propTypes = {
    humansAreLoaded: PropTypes.bool.isRequired,
    loadHumans: PropTypes.func.isRequired,
    loadLoremIpsum: PropTypes.func.isRequired,
    loremIpsumIsLoaded: PropTypes.bool.isRequired,
    humans: PropTypes.arrayOf(PropTypes.shape),
    loremIpsum: PropTypes.string,
    selectHuman: PropTypes.func,
    selectedHuman: PropTypes.shape({
      name: PropTypes.string,
    }),
  }

  static defaultProps = {
    humans: [],
    loremIpsum: 'Loading...',
    selectHuman: () => false,
    selectedHuman: undefined,
  }

  constructor(props) {
    super(props);
    this.selectHuman = this.selectHuman.bind(this);
  }

  componentWillMount() {
    const { humansAreLoaded, loremIpsumIsLoaded } = this.props;

    if (!humansAreLoaded) {
      this.props.loadHumans();
    }

    if (!loremIpsumIsLoaded) {
      this.props.loadLoremIpsum();
    }
  }

  selectHuman(indexToSelect) {
    this.props.selectHuman(indexToSelect);
  }

  render() {
    const { humans, loremIpsum, selectedHuman } = this.props;

    return (
      <Wrapper>
        {selectedHuman &&
          <HumanModal
            age={selectedHuman.age}
            bio={loremIpsum}
            birthday={selectedHuman.birthday}
            email={selectedHuman.email}
            imageUrl={selectedHuman.photo}
            onClick={this.selectHuman}
            phone={selectedHuman.phone}
            name={selectedHuman.name}
            surname={selectedHuman.surname}
          />}
        {humans.length ? humans.map(({ email, photo }, index) => (
          <Card
            index={index}
            key={email}
            onClick={this.selectHuman}
            imageUrl={photo}
          />)) :
        <Heading>Loading you some humans</Heading>}
      </Wrapper>
    );
  }
}

export function mapStateToProps({ humans, loremIpsum }) {
  return {
    humans: humans.list,
    humansAreLoaded: humans.isLoaded,
    indexOfSelectedHuman: humans.indexOfSelectedHuman,
    loremIpsumIsLoaded: loremIpsum.isLoaded,
    loremIpsum: loremIpsum.content[0],
    selectedHuman: humans.list[humans.indexOfSelectedHuman],
  };
}

export default connect(
  mapStateToProps,
  { loadHumans,
    loadLoremIpsum,
    selectHuman,
  })(UnconnectedHumans);
