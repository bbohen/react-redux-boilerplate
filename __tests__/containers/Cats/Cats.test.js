import React from 'react';

import { mapStateToProps, UnconnectedCats } from '../../../src/containers/Cats';
import testRendererWithProvider from '../../testUtils/testRendererWithProvider';

const mockFunction = jest.fn();
const mockCats = [
  {
    data: {
      id: 'foo1',
      thumbnail: 'cat-href-1',
    },
  },
  {
    data: {
      id: 'foo2',
      thumbnail: 'cat-href-2',
    },
  },
  {
    data: {
      id: 'foo3',
      thumbnail: 'cat-href-3',
    },
  },
];

describe('<UnconnectedCats />', () => {
  it('renders an empty cats container', () => {
    const component = testRendererWithProvider(<UnconnectedCats
      catsAreLoaded={false}
      loadCats={() => false}
    />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders some cats in a cats container', () => {
    const component = testRendererWithProvider(<UnconnectedCats
      cats={mockCats}
      loadCats={() => false}
      catsAreLoaded
    />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('tries to get you cats if you have none', () => {
    testRendererWithProvider(<UnconnectedCats
      catsAreLoaded={false}
      loadCats={mockFunction}
    />);

    expect(mockFunction.mock.calls.length).toBe(1);
  });

  it('should map state to props properly', () => {
    const props = mapStateToProps({
      cats: {
        isLoaded: true,
        list: mockCats,
      },
    });

    expect(props).toMatchSnapshot();
  });
});
