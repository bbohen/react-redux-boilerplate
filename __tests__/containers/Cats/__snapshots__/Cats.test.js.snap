// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UnconnectedCats /> renders an empty cats container 1`] = `
<div
  className="csKSle"
>
  <h2
    className="bhiNsp"
  >
    Loading you some cats
  </h2>
</div>
`;

exports[`<UnconnectedCats /> renders some cats in a cats container 1`] = `
<div
  className="csKSle"
>
  <div
    className="jkwvgr"
    onClick={[Function]}
  >
    <img
      alt=""
      className="bMAZFf"
      src="cat-href-1"
    />
  </div>
  <div
    className="jkwvgr"
    onClick={[Function]}
  >
    <img
      alt=""
      className="bMAZFf"
      src="cat-href-2"
    />
  </div>
  <div
    className="jkwvgr"
    onClick={[Function]}
  >
    <img
      alt=""
      className="bMAZFf"
      src="cat-href-3"
    />
  </div>
</div>
`;

exports[`<UnconnectedCats /> should map state to props properly 1`] = `
Object {
  "cats": Array [
    Object {
      "data": Object {
        "id": "foo1",
        "thumbnail": "cat-href-1",
      },
    },
    Object {
      "data": Object {
        "id": "foo2",
        "thumbnail": "cat-href-2",
      },
    },
    Object {
      "data": Object {
        "id": "foo3",
        "thumbnail": "cat-href-3",
      },
    },
  ],
  "catsAreLoaded": true,
}
`;
